lxd/network/driver_bridge.go

```diff
@@ -2543,6 +2543,33 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 	externalSubnets = append(externalSubnets, bridgeNetworkExternalSubnets...)
 	externalSubnets = append(externalSubnets, bridgedNICExternalRoutes...)
 
+	// Detect if there are any conflicting proxy devices on all instances with the to be created network forward
+	err = n.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		devices := db.ExpandInstanceDevices(inst.Devices, inst.Profiles)
+
+		for devName, devConfig := range devices {
+			if devConfig["type"] != "proxy" {
+				continue
+			}
+
+			proxyListenAddr, err := ProxyParseAddr(devConfig["listen"])
+			if err != nil {
+				return err
+			}
+
+			for netFwdAddr := range projectNetworksForwardsOnUplink {
+				if net.ParseIP(proxyListenAddr.Address).Equal(net.ParseIP(netFwdAddr)) {
+					return fmt.Errorf("Listen address %q conflicts with proxy device %q on %q in project %q", netFwdAddr, devName, inst.Name, p.Name)
```

Comment:
I don't think we should return an error here, as this function is just for returning "external subnets in use", not for casting a judgement on whether they conflict or not (we dont know that the caller of the function even wants to check for that).

Instead we should add the parsed proxy listen addresses to the `externalSubnets` slice with new `subnetUsageType` type (https://github.com/lxc/lxd/blob/7b3f46e08688043134b05888bfce9860e7274bcf/lxd/network/driver_common.go#L56-L65) called `subnetUsageProxy`.
See

Reply:
Like we do here https://github.com/lxc/lxd/blob/7b3f46e08688043134b05888bfce9860e7274bcf/lxd/network/driver_common.go#L1105-L1119 but filling the instance fields on that struct instead.